components/post_view/commented_on/commented_on.jsx

```diff
@@ -59,14 +54,18 @@ export default class CommentedOn extends PureComponent {
     render() {
         const username = this.makeUsername();
         const message = this.makeCommentedOnMessage();
+        const parentPostUser = this.props.parentPostUser;
+        const parentPostUserId = parentPostUser && parentPostUser.id;
 
-        const name = (
-            <a
-                className='theme'
-                onClick={this.handleOnClick}
-            >
-                {username}
-            </a>
+        const parentUserProfile = (
+            <UserProfile
+                user={parentPostUser}
+                userId={parentPostUserId}
+                displayName={username}
```

Comment:
wouldn't this username be related to the current post instead of the parent's?
also, looks like this one is being provided by [mapping the state](https://github.com/mattermost/mattermost-webapp/blob/master/components/user_profile/index.js#L15), maybe you wanted to set `displayUserName` to true so it uses the name from the `user` object?

Reply:
@rakhi2104 Do you have thoughts on Guillermo's questions above? We'd be glad to help!

Reply:
@rakhi2104 let me know if you want to discuss this either here or in the community server :)

Reply:
from community's conversation: 
> The CommentedOn's makeUsername generates the Original post's username

> The User object will be fetched asynchronously, and will not have username at the point when rendered, so it should be passed explicitly.